types/node/index.d.ts

```diff
@@ -941,10 +907,11 @@ declare module "http" {
         [errorCode: number]: string;
         [errorCode: string]: string;
     };
+
     export function createServer(requestListener?: (request: IncomingMessage, response: ServerResponse) => void): Server;
     export function createClient(port?: number, host?: string): any;
-    export function request(options: RequestOptions | string | URL, callback?: (res: IncomingMessage) => void): ClientRequest;
-    export function get(options: RequestOptions | string | URL, callback?: (res: IncomingMessage) => void): ClientRequest;
+    export function request(options: string | RequestOptions, callback?: (res: IncomingMessage) => void): ClientRequest;
```

Comment:
Currently it's redundant as it is enough that at least one property from `URL` matches `RequestOptions`. But `URL` would be also allowed if there would be no match as this is a separate path in code. Was added somewhere in NodeJS 7.x